src/com/opc/opcconnection.cpp

```diff
@@ -52,8 +52,34 @@ int COpcConnection::removeGroupCallbackDesc(const std::string& paGroupName){
     if(mOpcGroupMapList[i]->mGroupName == paGroupName){
       mEventHandler->removeComCallback(mOpcGroupMapList[i]->mCallbackDesc);
       DEVLOG_INFO("erase from m_lOpcGroupMapList %d[%s]\n", i,paGroupName.c_str());
-      delete mOpcGroupMapList[i];
+      //remove group's read item list first
+      for(auto &readItem : mOpcGroupMapList[i]->mReadItemsList){
+        if(nullptr != readItem){
+          delete readItem;
+          readItem = nullptr;
```

Comment:
as you are clearing the list with the swap below you don't need to set the item to nullptr. This applies to all other code below as well.